data_structures/abstract_hash_table.py

```diff
@@ -1,17 +1,36 @@
 from __future__ import annotations
 from abc import ABC, abstractmethod
-from typing import TypeVar, Generic, Tuple
+from typing import TypeVar, Generic, Tuple, List
 from data_structures.referential_array import ArrayR
+from data_structures.dunder_protected import DunderProtected
 
 K = TypeVar('K')
 V = TypeVar('V')
 
 
-class HashTable(ABC, Generic[K, V]):
+class HashTable(ABC, Generic[K, V], DunderProtected):
     """
     Hash Table (Map/Dictionary) ADT. 
     """
 
+    __TABLE_SIZES = [5, 13, 29, 53, 97, 193, 389, 769, 1543, 3079, 6151, 12289, 24593, 49157, 98317, 196613, 393241, 786433, 1572869]
+
+    def __init__(self, sizes: None | List[int] = None, hash_base: int | None = 31) -> None:
+        """
+        :param sizes: Optional list of sizes to use for the hash table.
+                      If not provided, a default list of sizes will be used.
+        :complexity: O(1) - Assuming the default sizes are used, we can assume the array is created in O(1) time.
+            If you use this function in any way that passes some variable input for the sizes, then the complexity
+            needs to change accordingly.
+        """
+        if sizes is not None:
+            self.__TABLE_SIZES = sizes
+
+        self.__size_index = 0
+        self.__array: ArrayR[tuple[str, V]] = ArrayR(self.__TABLE_SIZES[self.__size_index])
+        self.__length = 0
+        self.__hash_base = hash_base
+
```

Comment:
I would leave this in `LinearProbeTable`. It's implementation details and it only applies to linear probing. If it stays there it would still be inheritted by DoubleHashing and QuadraticProbing, without finding its way to SeparateChaining (which also inherits from this abstract class).